### PR TITLE
Allow proxies to be supplied to the client

### PR DESCRIPTION
**Pull request by kjaym**
In case users have to proxy through a URL, this adds support for including the proxy list in a CitrinationClient instance.

### CODE REVIEW

File: citrination_client/client.py [hunk 1]
```diff
@@ -17,20 +17,23 @@ class CitrinationClient(object):
     Class for interacting with the Citrination api.
     """
 
-    def __init__(self, api_key, site='https://citrination.com'):
+    def __init__(self, api_key, site='https://citrination.com', proxies=None):
```
Comment (sinistral) [inline citrination_client/client.py]: Why not have the default be an empty map rather than `None`?  It'll save having to do the explicit check later on.

Comment (kjaym) [reply citrination_client/client.py]: I would want to keep the check in case a user passed in `proxies=None`

Comment (sinistral) [reply citrination_client/client.py]: Does that actually have consequences?  Does it `requests` to pass `None` rather than empty map?

Comment (kjaym) [reply citrination_client/client.py]: It turns out that both `{}` and `None` work just fine with requests. I just removed the if/else check and just pass through the user input, defaulting to None.

**Comment by kjaym** [comment]
@sinistral We have tested configuring requests directly with a proxy.

**Pull request merged**